Sim/SimDelphesInterface/src/DelphesSimulation.cpp

```diff
@@ -343,78 +369,452 @@ StatusCode DelphesSimulation::finalize() {
+        particle.EndVertex(colMCParticles->at(idPartEndVertex).StartVertex());
+      }
+      else {
+        fcc::Point point;
+        point.X  = daughter->Position.X();
+        point.Y  = daughter->Position.Y();
+        point.Z  = daughter->Position.Z();
+
+        auto vertex = colGenVertices->create();
+        vertex.Position(point);
+        vertex.Ctau(cand->Position.T());
+        particle.EndVertex(vertex);
+      }
+      for (int iDaughter=cand->D1; iDaughter<=cand->D2; iDaughter++) {
+        if (iDaughter>=0 && m_vecPartProdVtxIDDecVtxID[iDaughter].second==-1) m_vecPartProdVtxIDDecVtxID[iDaughter].first = j;
+      }
+    }
+    // Test print
+    /*std::cout << "Delphes MCParticle: "
+              << " Id: "       << setw(3)  << j+1
+              << " Pdg: "      << setw(5)  << particle.Core().Type
+              << " Stat: "     << setw(2)  << particle.Core().Status
+              << " Bits: "     << setw(2)  << particle.Core().Bits
+              << " Px: "       << setw(10) << particle.Core().P4.Px
+              << " Py: "       << setw(10) << particle.Core().P4.Py
+              << " Pz: "       << setw(10) << particle.Core().P4.Pz
+              << " E: "        << setw(10) << sqrt(particle.Core().P4.Px*particle.Core().P4.Px + particle.Core().P4.Py*particle.Core().P4.Py + particle.Core().P4.Pz*particle.Core().P4.Pz + particle.Core().P4.Mass*particle.Core().P4.Mass)
+              << " M: "        << setw(10) << particle.Core().P4.Mass;
+    if (particle.StartVertex().isAvailable()) {
+    std::cout << " VId: "      << setw(3)  << particle.StartVertex().getObjectID().index
+              << " Vx: "       << setw(10) << particle.StartVertex().Position().X
+              << " Vy: "       << setw(10) << particle.StartVertex().Position().Y
+              << " Vz: "       << setw(10) << particle.StartVertex().Position().Z
+              << " T: "        << setw(10) << particle.StartVertex().Ctau();
+    }
+    if (particle.EndVertex().isAvailable()) {
+    std::cout << " VId: "      << setw(3)  << particle.EndVertex().getObjectID().index
+              << " Vx: "       << setw(10) << particle.EndVertex().Position().X
+              << " Vy: "       << setw(10) << particle.EndVertex().Position().Y
+              << " Vz: "       << setw(10) << particle.EndVertex().Position().Z
+              << " T: "        << setw(10) << particle.EndVertex().Ctau();
+    }
+    std::cout << std::endl;*/
+  }
+}   
 
-  Candidate * cand;
-  for(int j = 0; j < Input->GetEntries(); ++j) {
+void DelphesSimulation::ConvertTracks(const TObjArray*  Input,
+                                      fcc::ParticleCollection*   colParticles,
+                                      fcc::MCParticleCollection* colMCParticles,
+                                      fcc::ParticleMCParticleAssociationCollection* ascColParticlesToMC)
+{
+  Candidate* cand = nullptr;
+  for(int j=0; j<Input->GetEntries(); j++) {
 
     cand = static_cast<Candidate *>(Input->At(j));
```

Comment:
I don't understand why you need these static_casts here and at other places. What's the return-type before casting?